src/Features/Core/Portable/EditAndContinue/Utilities/Extensions.cs

```diff
@@ -183,9 +183,33 @@ private static bool ParsePrimaryParameterBackingFieldName(string fieldName, [Not
         => (IMethodSymbol?)constructor.ContainingType.GetMembers(WellKnownMemberNames.DeconstructMethodName).FirstOrDefault(
             static (symbol, constructor) => symbol is IMethodSymbol method && HasDeconstructorSignature(method, constructor), constructor)?.PartialAsImplementation();
 
-    // https://github.com/dotnet/roslyn/issues/73772: does this helper need to be updated to use IPropertySymbol.PartialImplementationPart?
+    /// <summary>
+    /// Returns a partial implementation part of a partial member, or the member itself if it's not partial.
+    /// </summary>
     public static ISymbol PartialAsImplementation(this ISymbol symbol)
-        => symbol is IMethodSymbol { PartialImplementationPart: { } impl } ? impl : symbol;
+        => PartialImplementationPart(symbol) ?? symbol;
+
+    public static bool IsPartialDefinition(this ISymbol symbol)
+        => symbol is IMethodSymbol { IsPartialDefinition: true } or IPropertySymbol { IsPartialDefinition: true };
+
+    public static bool IsPartialImplementation(this ISymbol symbol)
+        => symbol is IMethodSymbol { PartialDefinitionPart: not null } or IPropertySymbol { PartialDefinitionPart: not null };
+
+    public static ISymbol? PartialDefinitionPart(this ISymbol symbol)
+        => symbol switch
+        {
+            IMethodSymbol { PartialDefinitionPart: var def } => def,
+            IPropertySymbol { PartialDefinitionPart: var def } => def,
+            _ => symbol
```

Comment:
nit: it feels like this helper should return null if we couldn't even convert the symbol to something with a `PartialDefinitionPart` property

Reply:
Good catch.